### PR TITLE
Implement new design for "more on gov.uk" section of the homepage

### PR DESCRIPTION
**Pull request by owenatgov**
## What
Implements the new homepage design for the "More on gov.uk" section of the homepage. Additionally fixes a minor CSS bug on the `grid-helper` sass helper.

## Why
Part of GOV.UK Explores mission to set the tone for new incoming page designs across govuk.

[Card](https://trello.com/c/0M4lPxP0/645-homepage-more-on-govuk), [Jira issue NAV-3300](https://gov-uk.atlassian.net/browse/NAV-3300)

[Test URL](https://govuk-fronte-new-homepa-67k9kq.herokuapp.com/)

### Note
I haven't removed the horizontal rule above this section. I propose we tackle this along with the other borders between sections once we implement the whole page in https://github.com/alphagov/frontend/pull/3027

### Desktop
Before:
Screenshot 2021-11-26 at 13 54 38

After:
Screenshot 2021-11-26 at 13 54 18

### Tablet (700px width)
Before:
Screenshot 2021-11-26 at 13 55 29

After:
Screenshot 2021-11-26 at 13 55 15

### Mobile (380px width)
Before:
Screenshot 2021-11-26 at 13 56 06

After:
Screenshot 2021-11-26 at 13 55 53

### States for Bank Holidays box
Previously this wasn't interactable to assistive tech users so there are no "before" screenshots here

Hover:
Screenshot 2021-11-26 at 14 05 06

Focus:
Screenshot 2021-11-26 at 13 56 40

### CODE REVIEW

**Comment by mia-allers-gds** [comment]
@owenatgov looks good! Is there any way we could constrain the width of the  'bank holidays' block at around 640px?
Screenshot 2021-11-26 at 14 49 08

File: app/views/homepage/_more_on_govuk.html.erb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+          data-track-category="homepageClicked"
+          data-track-action="promoImageLink"
+          data-track-label="/bank-holidays"
+          data-track-value="1"
+          data-track-dimension="<%= t('homepage.index.uk_bank_holidays') %>"
+          data-track-dimension-index="29">
+            <%= t'homepage.index.uk_bank_holidays' %>
+        </a>
+      </h3>
+    </div>
```
Comment (jon-kirwan) [inline app/views/homepage/_more_on_govuk.html.erb:50]: Suggestion from @injms on a separate PR. You can use a heading component like so -
```
<%= render "govuk_publishing_components/components/heading", {
  font_size: "m",
  heading_level: 3,
  margin_bottom: 2,
  text: link_to(item[:title], "/travel-abroad", {
    class: "govuk-link",
    data: {
      track_category: "homepageClicked",
      track_action: "promoTextLink",
      track_label: "/travel-abroad",
      track_value: 1,
      track_dimension-index: 29,
      track_dimension: item[:title],
    }
  }),
} %>
```
https://github.com/alphagov/frontend/pull/3029#discussion_r756998218

File: app/views/homepage/index.html.erb [hunk 2]
```diff
@@ -98,159 +98,6 @@
 
     <hr class="home-divider" id="more-on-govuk" aria-hidden="true" />
```
Comment (jon-kirwan) [inline app/views/homepage/index.html.erb:100]: Question on the `hr` and whether it's better to change these to `border-top`s now (or in `new-homepage`)

File: app/views/homepage/_more_on_govuk.html.erb [hunk 3]
```diff
@@ -0,0 +1,52 @@
+<section class="homepage-more-on-govuk">
+  <%= render "govuk_publishing_components/components/heading", {
+    text: t('homepage.index.more'),
+    margin_bottom: 6,
+    font_size: "m",
+  } %>
+  <%= render "govuk_publishing_components/components/heading", {
+    text: t('homepage.index.most_active'),
+    margin_bottom: 4,
```
Comment (jon-kirwan) [inline app/views/homepage/_more_on_govuk.html.erb:9]: Bit of a nitpick here but I think the translate strings (`homepage.index.more` and `homepage.index.most_active` are wrapped in double quotes elsewhere.

File: app/views/homepage/_more_on_govuk.html.erb [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        <a href="/bank-holidays"
+          class="govuk-link homepage-promo__link"
+          data-track-category="homepageClicked"
+          data-track-action="promoImageLink"
+          data-track-label="/bank-holidays"
+          data-track-value="1"
+          data-track-dimension="<%= t('homepage.index.uk_bank_holidays') %>"
+          data-track-dimension-index="29">
+            <%= t'homepage.index.uk_bank_holidays' %>
+        </a>
```
Comment (jon-kirwan) [inline app/views/homepage/_more_on_govuk.html.erb:48]: Ah same down here too - so `homepage.index.uk_bank_holidays` and `homepage.index.uk_bank_holidays` might need to be changed to double quotes.

**Review by jon-kirwan — CHANGES_REQUESTED** [review]

File: app/views/homepage/index.html.erb (hunk 2 above)
Comment (owenatgov) [reply app/views/homepage/index.html.erb:100]: I reckon personally let's do it in the full merge. That way we can see everything all at once and more effectively make decisions about the borders/spacing. If you'd really prefer I remove this now then I'll remove it now.

File: app/views/homepage/_more_on_govuk.html.erb (hunk 1 above)
Comment (owenatgov) [reply app/views/homepage/_more_on_govuk.html.erb:50]: I'm going to push back on this as this technically isn't a suitable use for the heading component. It's being styled as a blue box but marked up as a heading for screen reader users.

**Comment by owenatgov** [comment]
@mia-allers-gds Made the changes to the blue box on mobile as requested. Will update the screenshots in just a jiffy but the  change can be seen [here](https://govuk-fronte-new-homepa-67k9kq.herokuapp.com/).

File: app/views/homepage/index.html.erb (hunk 2 above)
Comment (jon-kirwan) [reply app/views/homepage/index.html.erb:100]: No problem, agreed it makes more sense to revisit this after the full merge.

**Review by jon-kirwan — APPROVED** [review]
Thanks Owen, all looks good now.

There is a failing test which you've probably already seen -
```
HomepageTest#test_: when visiting a Welsh content item first should use the English locale after visiting the Welsh content.  [/var/lib/jenkins/workspace/end_new-homepage--more-on-gov.uk/test/integration/homepage_test.rb:37]:
Expected false to be truthy.

rails test test/integration/homepage_test.rb:34
```
Otherwise all looks fine so have approved 💥

**Comment by mia-allers-gds** [comment]
@owenatgov looks much better, thank you! Good to go :)

**Pull request merged**